src/websocket/websocket_secure_endpoint.cpp

```diff
@@ -0,0 +1,83 @@
+#include "./websocket_secure_endpoint.h"
+#include "../logging.h"
+
+namespace sensr
+{   
+  WebSocketSecureEndPoint::WebSocketSecureEndPoint(const std::string& cert_path) : 
+    WebSocketEndPointBase(),
+    cert_path_(cert_path)
+  {
+    Init(endpoint_);
+  }
+
+  WebSocketSecureEndPoint::~WebSocketSecureEndPoint() {
+    Fin(endpoint_);
+  }
+
+  bool WebSocketSecureEndPoint::Connect(const std::string &uri,
+                 const WebSocketEndPointBase::MsgReceiver& func,
+                 const WebSocketEndPointBase::ErrorReceiver& err_func) {
+    
+    if (!connection_hdl_.expired()) {
+      INFO_LOG(uri + " is already connected.");
+      return true;
+    }
+    bool ret = false;
+    try {
+      if (!cert_path_.empty()) { 
+        endpoint_.set_tls_init_handler(std::bind(
+          &WebSocketSecureEndPoint::OnTLSInit,
+          this,
+          std::placeholders::_1));
```

Comment:
Can I keep this as std::bind?
For now `OnTLSInit` is quite short function. but it can be expanded later depending on requirement.
I worry that lambda can decrease readability in this situation.

Reply:
I can not really understand why it does decrease readability. But it is not big deal. I am fine with leaving it as `std::bind` :D

Reply:
Ah, I meant `Connect()` function. For now, this class register few websocket handler only(Frankly, I have not much idea on websocket, so I just registered minimum handlers only.) And, if we add more websocket handler in the future, then there will be multiple lambdas are existed in `Connect()` function and I thought this can lead long scrolling and poor readability.
Anyway, thank you for understanding.